packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,302 @@
+
+The visibility of the dialog is controlled through the `hidden` prop whose its value should be a react state boolean provided from the consumer.
+
+[Documentation for v8 Dialog](https://developer.microsoft.com/en-us/fluentui#/controls/web/dialog)
+
+Sample code:
+
+```jsx
+  <DefaultButton secondaryText="Opens the Sample Dialog" onClick={toggleHideDialog} text="Open Dialog" />
+
+  <Dialog
+    hidden={hideDialog}
+    onDismiss={toggleHideDialog}
+    dialogContentProps={dialogContentProps}
+    modalProps={modalProps}
+  >
+    <DialogFooter>
+      <PrimaryButton onClick={toggleHideDialog} text="Send" />
+      <DefaultButton onClick={toggleHideDialog} text="Don't send" />
+    </DialogFooter>
+  </Dialog>
+
+```
+
+#### v0
+
+In v0, the Dialog component expects all the content through props, including the content, actions etc. The dialog component uses the `trigger` prop that expects a React component to control its visibility. The element passed to this prop will be rendered in-place where the dialog is defined.
+
+[Documentation for v0 Dialog](https://fluentsite.z22.web.core.windows.net/components/dialog/definition)
+
+```jsx
+<Dialog
+  cancelButton="Connect protocol"
+  confirmButton="Transmit capacitor"
+  content="Connect driver"
+  header="Transmit capacitor"
+  headerAction="Generate protocol"
+  trigger={<Button content="A trigger" />}
+/>
+```
+
+## API proposal
+
+Dialog component is composed by 3 sub-components: Header, Body and Footer.
+
+### Components
+
+| Component    | Purpose                                                  |
+| ------------ | -------------------------------------------------------- |
+| Dialog       | The main wrapper component.                              |
+| DialogHeader | (optional) Component for the title and the close button. |
+| DialogBody   | (optional) Component for the main content of dialog.     |
+| DialogFooter | (optional) Component for the main actions of dialog.     |
+
+## Dialog
+
+The dialog is a container which handles styling (border, background etc.) and the overlay (dimmed background).
+
+### Anatomy
+
+![Visual anatomy of Dialog component](./assets/dialog-anatomy.png)
+
+### API
+
+| Property    | Values                        | Default   | Purpose                                    |
+| ----------- | ----------------------------- | --------- | ------------------------------------------ |
+| type        | `modal`, `non-modal`, `alert` | `modal`   | Dialog variations                          |
+| isOpen      | boolean                       | `false`   | Set to `true` when the dialog is visible   |
+| isDraggable | boolean                       | `false`   | Set to `true` to make the dialog draggable |
```

Comment:
hmm not sure this is good idea as this would vendor lock user to our particular draggable implementation ( we dont have for v9 any ATM AFAIR ).

ideally this should be open to extension with any 3rd party solution. MUI has it nicely implemented https://mui.com/components/dialogs/#draggable-dialog

Reply:
Yes totally agree, this is also being discussed with the design team and there is a high chance that it will not be implemented within the dialog component. I will remove it from this table and add it as a pending issue until we have a clear answer about this.